(ДЗ) 1+2: Индикацию запуска колбэк-методов активности сделал - один и тот же метод пишет и в лог, и тосты.

udav2alex opened the pull request:
(ДЗ) 1+2: Индикацию запуска колбэк-методов активности сделал - один и тот же метод пишет и в лог, и тосты.
3: Для сохранения выбранного города использовал onSaveInstanceState и onRestoreInstanceState.
4: Создал синглтон (без синхронизации), который хранит единственную ссылку на объект WeatherState - текущую погоду. Перспектив для использования этого подхода пока не вижу, поэтому сильно глубоко не копал. Можно это дело оставить до тех пор, пока не появится нормальная сохранялся погодных данных (в БД, видимо).

Переделал работу с городом:
- добавил класс DestinationPoint, в котором инкапсулируется "точка назначения" с городом и райном (теперь точку назначения будет легко расширять)
- изменил окно выбора города - теперь в MainActivity передается не город, а точка назначения
- DestinationPoint сделал Parcelable, в onSaveInstanceState и onRestoreInstanceState сохраняется и восстанавливается посылка с точкой назначения

Начал работу над погодой:
- Создал класс WeatherState, в который буду инкапусулировать все погодные параметры: температуру, облачность, влажность, давление и т.п.

udav2alex committed:
(ДЗ) 1+2: Индикацию запуска колбэк-методов активности сделал - один и тот же метод пишет и в лог, и тосты.
3: Для сохранения выбранного города использовал onSaveInstanceState и onRestoreInstanceState.
4: Создал синглтон (без синхронизации), который хранит единственную ссылку на объект WeatherState - текущую погоду. Перспектив для использования этого подхода пока не вижу, поэтому сильно глубоко не копал. Можно это дело оставить до тех пор, пока не появится нормальная сохранялся погодных данных (в БД, видимо).

Переделал работу с городом:
- добавил класс DestinationPoint, в котором инкапсулируется "точка назначения" с городом и райном (теперь точку назначения будет легко расширять)
- изменил окно выбора города - теперь в MainActivity передается не город, а точка назначения
- DestinationPoint сделал Parcelable, в onSaveInstanceState и onRestoreInstanceState сохраняется и восстанавливается посылка с точкой назначения

Начал работу над погодой:
- Создал класс WeatherState, в который буду инкапусулировать все погодные параметры: температуру, облачность, влажность, давление и т.п.

DwarfPorter commented on app/src/main/java/ru/gressor/weatherapp/MainPresenter.java:
```diff
@@ -0,0 +1,27 @@
+
+    public static MainPresenter getInstance() {
+        if (instance == null) {
+            instance = new MainPresenter();
```
а почему не используете synchronize?

udav2alex committed:
(ДЗ4) 1: Уже было сделано. 2: Сделал запуск поиска в вебе - работает адекватно, показывает погоду сам гугл/яндекс, плюс можно выбрать подходящий сайт с подробностями.

Кроме этого:
- Поддержал в "снимке погоды" WeatherState интерфейс Parcelable
- Сделал сохранение/загрузку состояния погоды в onSaveInstanceState и onRestoreInstanceState, удалил код с синглтоном
- Перенес заполнение интерфейса данными в onRestore: когда восстанавливается интефейс после поворота экрана, onCreate срабатыват до восстановления данных в onRestoreInstanceState и приложение падает, т.к. нужный WeatherState еще пуст.
